docs/solutions/rhel74_image/datasources.tf

```diff
@@ -35,20 +35,35 @@ data "oci_core_subnets" "subnet" {
 
 }
 
+# Gets the OCID of the image. This technique is for example purposes only. 
+# The results of oci_core_images may change over time for Oracle-provided images, 
+# so the only sure way to get the correct OCID is to supply it directly.
+
+data "oci_core_images" "image" {
```

Comment:
Doesn't appear to be used anywhere. See https://github.com/oracle/terraform-provider-oci/pull/399/files, which switched to using the image OCID directly. That will be the recommended best practice going forward, so we should stick with that approach.

Reply:
The problem with the image OCID is that it is specific to the region, so the user of the template must specify the correct OCID for each region.  This method (which is modified in the latest commit) allows the specification of the OS name and version, and selects the first one on the list.  This allows for the correct OCID to be specified regardless of the region.

The image datasource is used in the ipxe.tf to specify the image to be used for the ipxe instance.

New commit with both OS name and version forthcoming

Reply:
OS name and version in current commit.